src/Rector/v9/v0/ReplaceExtKeyWithExtensionKeyRector.php

```diff
@@ -105,7 +114,29 @@ private function createExtensionKeyFromFolder(SmartFileInfo $fileInfo): ?SmartFi
     private function isAssignment(Variable $node): bool
     {
         $parentNode = $node->getAttribute(AttributeKey::PARENT_NODE);
+
         // Check if we have an assigment to the property, if so do not change it
         return $parentNode instanceof Assign && $parentNode->var === $node;
     }
+
+    private function resolveExtensionKeyByComposerJson(SmartFileInfo $extEmConf): ?string
+    {
+        try {
+            $composerJson = new SmartFileInfo($extEmConf->getRealPathDirectory() . '/composer.json');
+            $json = json_decode($composerJson->getContents(), true);
+
+            if (isset($json['extra']['typo3/cms']['extension-key'])) {
+                return $json['extra']['typo3/cms']['extension-key'];
+            }
+
+            if (isset($json['name'])) {
+                [, $extensionKey] = explode('/', $json['name'], 2);
```

Comment:
I would implement this differently. Usually the extension key is `vendor_ext`. Any when naming a package, users usually use `vendor/ext`. What you do here is drop the vendor completely and use the name only as possible folder name. This works for popular extensions like `news` and `powermail` where the uniqueness is given by the uniqueness inside the TER but those are usually not the extensions you touch during an update in your instance. It's your own local extensions which are mostly prefixed with a customer or agency prefix aka the vendor. Hence, I would try to combine both parts of the name. vendor and package name.